client/pages/edit-entry-form.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+
+class EditForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entry: null,
+      categoryList: [],
+      amount: null,
+      date: null,
+      description: null,
+      categoryId: null
+    };
+
+    this.handleChangeCategory = this.handleChangeCategory.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeAmount = this.handleChangeAmount.bind(this);
+    this.handleChangeDescription = this.handleChangeDescription.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/entry/${this.props.entryId}`)
+      .then(response => response.json())
+      .then(entry => this.setState({ entry }));
+
+    fetch('/api/category-table')
+      .then(response => response.json())
+      .then(categoryList => this.setState({ categoryList }));
+  }
+
+  handleChangeCategory(event) {
+    const newEntry = this.state.entry;
+    newEntry.category = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleChangeDate() {
+    const newEntry = this.state.entry;
+    newEntry.date = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleChangeAmount() {
+    const newEntry = this.state.entry;
+    newEntry.amount = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleChangeDescription() {
+    const newEntry = this.state.entry;
+    newEntry.description = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleClick() {
+    const req = {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        amount: this.state.entry.amount,
+        date: this.state.entry.date,
+        description: this.state.entry.description,
+        categoryId: this.state.entry.category
+      })
+    };
+    fetch(`/api/update-entry/${this.props.entryId}`, req);
+  }
+
+  render() {
+
+    const entry = this.state.entry;
+    if (!entry) {
+      return <div>LOADING...</div>;
+    }
+
+    const dateComponents = entry.date.split('/');
+    if (dateComponents.length > 1) {
+      entry.date = `${dateComponents[2]}-${dateComponents[0]}-${dateComponents[1]}`;
+    }
+
+    const categoryList = this.state.categoryList;
+    const categoryOptions = categoryList.map(category => {
+      return (
+        <option key={category.categoryId} value={category.categoryId}>{category.name}</option>
+      );
+    });
+
+    let categoryId = null;
+
+    for (let i = 0; i < categoryList.length; i++) {
+      if (entry.category === categoryList[i].name) {
+        categoryId = categoryList[i].categoryId;
+      }
+    }
```

Comment:
Instead of comparing the name, just send the `categoryId` to the client from the server along with the category name.